### DISCUSSION TITLE
Unable to pass another container's network through the nordlynx container

### DISCUSSION THREAD

**Discussion by imthenachoman** [bubuntux/nordlynx#45] [Q&A] [answered]
I've got nordlynx running in a container on unRAID (screenshot below). I've configured another container to use the nordlynx container for the network -- and it works too. Both containers show my VPN IP when I connect to the container's shell and use `curl ifconfig.io`.

I've configured the port mappings as I'm supposed to but I still cannot connect the other container's web UI.

- In the nordlynx container:
  - I've got multiple VLANs on my home network so I used `192.168.1.0/16` for `NET_LOCAL`
  - I want to map `8989` to `8989` so I can access Sonarr
- In the Sonarr container:
  - I am using `--net=container:nordlynx`
  - The web UI is accessed on `8989`

Can anyone please help me with what is wrong?

![image](https://user-images.githubusercontent.com/83817/155611721-d781b93d-3a4c-42ef-b460-8b52205fab02.png)

![image](https://user-images.githubusercontent.com/83817/155612119-37c6fcaa-683b-4c68-b8a0-6bdfac45289f.png)

**Comment by ginodesilva** [comment]
I haven't got much experience with unRAID but for the vpn service you have to map the port. So instead of 8989, try 8989:8989?

**Reply by imthenachoman** [reply]
Yeah, it is mapped like that. But oddly enough, it's working now.

**Comment by imthenachoman** [comment, marked as answer]
I have no clue what happened but all of a sudden it is working. I want to assume something was wrong with the images and they just had to be rebuilt. :/

**Comment by thomasedain** [comment]
I am experiencing the exact same issue. Everything is mapped in the NordLynx container port wise, but I cannot access the mapped services. If I check in the console for each of those services, I can see that traffic is being routed via the NordLynx container correctly. (checked with curl ifconfig.io)

I used NordVPN container for a long time, but never experienced this issue.  I am abit stuck with how I can make this work.

**Reply by imthenachoman** [reply]
Try rebuilding all of the container images.

**Comment by neildeadman** [comment]
Not sure if this is similar to what I see... basically if I down my entire stack and up it with no containers running, it all works. If one of the client containers restarts, then I can no longer connect. 
I initially thought it wasn't working at all when I first started out, but then it just started. I pinned it down to the above. I've come here today to find a solution hopefully.

**Reply by neildeadman** [reply]
Sorry, I've just done further testing and the above isn't correct :(

It seems if the VPN container restarts, then I can no longer connect to the services using the VPN containers network. I have just found that if I restart the client service containers I can reconnect. So that is something, and possibly able to script a working solution for that scenario.

**Reply by imthenachoman** [reply]
I think I am noticing the same thing. I need to investigate/research this.

**Reply by neildeadman** [reply]
I've been reading these:

https://github.com/docker/compose/issues/6626

https://github.com/wfg/docker-openvpn-client/issues/8

**Comment by shiftylilbastrd** [comment]
Recently had to rebuild my Docker image after an Unraid cache swap. Now for some reason I can only access the WebUI of one of the three containers I have routed through NordLynx. All three are configured the same as they were previously so I really don't understand what's happening. Restarting the containers/stack causes no changes.
